src/device/prims_simple.h

```diff
@@ -197,14 +198,37 @@ class Primitives<
     }
   }
 
+  #if defined(__gfx942__) && defined(HIP_UNCACHED_MEMORY)
+  #define RCCL_USE_CHEAPER_THREADFENCE_POSSIBLE 1
+  #else
+  #define RCCL_USE_CHEAPER_THREADFENCE_POSSIBLE 0
+  #endif 
+
+  template<bool UseCheaperThreadFence>
+  inline __device__ void gfx9ThreadFence();
+
+  template<>
+  inline __device__ void gfx9ThreadFence<true>() {
+    // __threadfence compiles to buffer_wbl2 sc1 plus buffer_inv sc1.  It turns out in this
+    // case we only need the buffer_inv sc1 due to the memory model we use.
+    asm volatile("buffer_inv sc1");
```

Comment:
pls provide all documentation from the discussion below on the assembly line you end up adding